### PR TITLE
feat(ui): adding path to the user's code block

### PR DESCRIPTION
**Pull request by wwayne**

### CODE REVIEW

**Comment by liangfung** [comment]
I've fixed the git_url issue in selectContext

**Comment by wwayne** [comment]
@liangfung ready for review
I removed this feature from VSCode (adding from: string) to the chat context

File: ee/tabby-ui/components/chat/question-answer.tsx [hunk 1]
```diff
@@ -99,6 +106,17 @@ function UserMessageCard(props: { message: UserMessage }) {
     return `\n${'```'}${language}\n${selectContext?.content ?? ''}\n${'```'}\n`
   }, [selectContext])
 
+  let selectCode: SelectCode | null = null
+  if (selectCodeSnippet && message.selectContext) {
+    const { range, filepath } = message.selectContext
+    const pathSegments = filepath.split('/')
+    selectCode = {
+      fileName: pathSegments[pathSegments.length - 1],
+      path: pathSegments.slice(0, pathSegments.length - 1).join('/'),
```
Comment (liangfung) [inline ee/tabby-ui/components/chat/question-answer.tsx:115]: It seems unnecessary to split the filepath into filename and path now.

Comment (wwayne) [reply ee/tabby-ui/components/chat/question-answer.tsx]: Updated, do I need to add `/` to the beginning?

**Review by liangfung — APPROVED** [review]

**Pull request merged**